contents/csharp/IntroductionToCSharp.md

```diff
@@ -0,0 +1,186 @@
+(and 10,000 globally), with an [average base pay of nearly $80,000](https://www.glassdoor.com/Salaries/c-net-developer-salary-SRCH_KO0,15.htm).
+
+## Glimpse of C# Syntax
+
+C# is a relatively high level language and is not difficult to pick up (a little more difficult than Python and a little easier than Java.)
+
+Below is a sample code snippet of what a simple program in C# might look like:
+
+```csharp
+//Comments can be marked with // or /**/
+//Namespaces are similar to packages, except the file does not need to be physically in the directory
+using ProjectName.Utils;
+
+namespace ProjectName.Model
+{
+    //Subclassing BaseClass
+    class MyClass : BaseClass {
+        ...
+        //Method declaration
+        public static void Main(String[] args) {
+            //Variable definition
+            string message = "Hello World!";
+            Console.WriteLine(message);
+        }
+    }
+}
+```
+
+[Click here to get started](#how-to-get-started)
+
+## Some Noteworthy Features of C#
+
+Most programming languages/frameworks have their own unique quirks, which can be utilised for better code quality.
+
+### Object / Array / Collection Initializers
+
+In C#, you can construct an [Object](https://en.wikipedia.org/wiki/Object_(computer_science) "In computer science, an object can be a variable, a data structure, a function, or a method, and as such, is a value in memory referenced by an identifier.")
+, [Array](https://www.webopedia.com/TERM/A/array.html "In programming, a series of objects all of which are the same size and type. Each object in an array is called an array element. For example, you could have an array of integers or an array of characters or an array of anything that has a defined data type.") 
+or [Collection](https://computersciencewiki.org/index.php/Collections "A collection — sometimes called a container — is simply an object that groups multiple elements into a single unit.") conveniently as shown. This can be useful when writing tests, as test data will be easier to read,
+as compared to calling the actual constructor.
+
+```csharp
+public class Foo {
+    Bar MyBar;
+}
+
+Bar bar = new Bar();
+Foo foo = new Foo() {
+    MyBar = bar;
+};
+```
+
+### Closures
+
+Sometimes, it may be useful to [defer execution](http://www.informit.com/articles/article.aspx?p=2171751 "Code that is executed only when results need to be evaluated. There are many reasons for executing code later") 
+or capture a local context for later execution. Context capturing is reflected below:
+
+```csharp
+//Capturing local context
+public class Context {
+    public static Func<Int> GetCounter() {
+        int count = 0;
+        return () => {
+            count++;
+            return count();
+        }
+    }
+}
+
+Func<Int> counter = Context.GetCounter();
+counter();
+```
```

Comment:
You need to explain this point in more details. The reader shouldn't have to go to external articles to be able to understand this article. External articles are for _additional_ info.